hw_2_functions.py

```diff
@@ -0,0 +1,284 @@
+# 1) Write a function that emulates the game "rock, scissors, paper"
+# At the entrance, your function accepts your version printed from the console, the computer makes a decision randomly.
+import random
+import pprint
+
+
+def get_input():
+    """Function returns options dictionary key for item selected by user via entering key or value"""
+    while True:
+        inp = input('Enter "rock" or 1, "scissors" or 2, "paper" or 3 to play the game\n')
+        if inp.isdigit():
+            if int(inp) in rsp_options.keys():
+                opt_key = int(inp)
+                return opt_key
+            else:
+                continue
+        elif inp.lower() in rsp_options.values():
+            opt_key = int(list(rsp_options.keys())[list(rsp_options.values()).index(inp.lower())])
+            return opt_key
+        else:
+            continue
+
+
+def define_winner(hum, comp):
+    # Define winner using win_loss_matrix
+    human, computer = hum-1, comp-1
+    win_loss_matrix = [[-1, 0, 2],
+                       [0, -1, 1],
+                       [2, 1, -1]]
+    rsp_winner = win_loss_matrix[human][computer]
+    if rsp_winner == human:
+        print("Congratulations! You WIN!")
+    elif rsp_winner == computer:
+        print("Sorry, computer wins..")
+    else:
+        print("DRAW GAME. You both selected the same items.")
+
+
+def play_game_rsp():
+    global rsp_options
+    rsp_options = {1: 'rock', 2: 'scissors', 3: 'paper'}
+    # Getting inputs from human and computer
+    user_move = get_input()
+    user_item = rsp_options[user_move]
+    computer_move = random.randint(1, 3)
+    comp_item = rsp_options[computer_move]
+    print(f'You selected "{user_item}". Computer selected "{comp_item}"')
+    # Defining the winner
+    define_winner(user_move, computer_move)
+
```

Comment:
It is cool that you try to separate different actions with the functions. However, it works for more complicated tasks. In this case, I think the code is overloaded with extra variables and data modifications that consume additional memory.